scripts/datasets/coco_tracking.py

```diff
@@ -0,0 +1,242 @@
+"""this script is used to prepare COCO dataset for tracking,
+which is 2017 COCO
+Code adapted from https://github.com/STVIR/pysot"""
+import argparse
+import zipfile
+from os.path import join, isdir, expanduser
+from concurrent import futures
+import sys
+import time
+import json
+import numpy as np
+from gluoncv.utils import download, makedirs
+from gluoncv.data.mscoco.utils import try_import_pycocotools
+from gluoncv.utils.filesystem import try_import_cv2
+
+def parse_args():
+    """COCO dataset parameter."""
+    parser = argparse.ArgumentParser(
+        description='Initialize MS COCO dataset.')
+    parser.add_argument('--download-dir', type=str, default='~/.mxnet/datasets/coco', help='dataset directory on disk')
+    parser.add_argument('--no-download', action='store_true', help='disable automatic download if set')
+    parser.add_argument('--overwrite', action='store_true', help='overwrite downloaded files if set, in case they are corrupted')
+    parser.add_argument('--instance-size', type=int, default=511,
+                        help='instance image size')
+    parser.add_argument('--num-threads', type=int, default=12,
+                        help='threads number')
+    args = parser.parse_args()
+    args.download_dir = expanduser(args.download_dir)
+    return args
+
+def download_coco(args, overwrite=False):
+    """download COCO dataset and Unzip to download_dir"""
+    _DOWNLOAD_URLS = [
+        ('http://images.cocodataset.org/zips/train2017.zip',
+         '10ad623668ab00c62c096f0ed636d6aff41faca5'),
+        ('http://images.cocodataset.org/annotations/annotations_trainval2017.zip',
+         '8551ee4bb5860311e79dace7e79cb91e432e78b3'),
+        ('http://images.cocodataset.org/zips/val2017.zip',
+         '4950dc9d00dbe1c933ee0170f5797584351d2a41'),
+    ]
+    if not isdir(args.download_dir):
+        makedirs(args.download_dir)
+    for url, checksum in _DOWNLOAD_URLS:
+        filename = download(url, path=args.download_dir, overwrite=overwrite, sha1_hash=checksum)
+        with zipfile.ZipFile(filename) as zf:
+            zf.extractall(path=args.download_dir)
+
+def printProgress(iteration, total, prefix='', suffix='', decimals=1, barLength=100):
```

Comment:
how is it different from `tqdm`?

Reply:
I think it's same from result. but  I have try to use tqdm replace printProgress and find tqdm that it shows every image time not all

Reply:
Would you please be clearer? As far as I know, `tqdm` prints something like `3664/100000 [00:11<05:09, 311.08it/s]` in the end of the progress bar, which reflects the passed time / rest time, and eventually total time once it finishes.

Reply:
I understand it.
if I use```
 for video in tqdm(videos):
            executor.submit(crop_video, args, sub_set, video, crop_path, ann_base_path)
```
replace
```
with futures.ProcessPoolExecutor(max_workers=args.num_threads) as executor:
       fs = [executor.submit(crop_video, args, sub_set, video, crop_path, ann_base_path) for video in videos]
```
will cause some problems.
Because the thread pool submits tasks asynchronously, the main thread does not wait for the thread created  that finish executing in the thread pool.From the results，The progress bar is over, but the process is not over。
so，use printProgress to  record when every process is over

Reply:
Good, sounds fair.